### PR TITLE
feat: Add profile fragment viewModel

### LINKED ISSUE #22: Create ViewModel for Profile Fragment
I would like to work on it
Parent issue https://github.com/fossasia/open-event-android-general/issues/21

### PR DESCRIPTION
**Pull request by nikit19**
Fixes #22

### CODE REVIEW

**Comment by simarsingh24** [comment]
LGTM 👍

**Comment by dreadpool2** [comment]
Looks great 👍

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 1]
```diff
@@ -35,50 +33,51 @@ class ProfileFragment : Fragment() {
...
             redirectToMain()
         }
 
+        profileFragmentViewModel.progress.observe(this, Observer {
+            it?.let { rootView.progressBar.isIndeterminate = it }
+        })
+
+        profileFragmentViewModel.visibility.observe(this, Observer {
+            it?.let { rootView.progressBar.visibility = it }
+        })
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Visibility and determinate should be combined. ViewModel should not know about the semantics of view

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally {
+                    progress.value = false
+                    visibility.value = View.GONE
+                }
+                .subscribe({ user ->
+                    Timber.d("Response Success")
+                    name.value = "${user.firstName} ${user.lastName}"
+                    email.value = user.email
+                    avatarUrl.value = user.avatarUrl
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Rather directly send the user object to view

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+                }) { throwable -> Timber.e(throwable, "Failure") })
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Send the error to view

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 1 above)
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: How should I combine these 2 mutableLiveData ?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: No need to combine LiveData, see how it is handled in Login and Event fragment

**Comment by nikit19** [comment]
@iamareebjamal Please review

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        compositeDisposable.add(authService.getProfile()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally {
+                    progress.value = false
+                }
+                .subscribe({ user ->
+                    Timber.d("Response Success")
+                    this.user.value = user
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Refrain from adding extra lines

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    fun isLoggedIn() = authService.isLoggedIn()
+
+    fun logout() = authService.logout()
+
+    fun fetchProfile() {
+        compositeDisposable.add(authService.getProfile()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Add doOnSubscribe to add enable progress

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                    progress.value = false
+                }
+                .subscribe({ user ->
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragmentViewModel.kt]: Move to previous line

**Comment by nikit19** [comment]
PR updated

**Pull request merged**